contracts/contracts/strategies/NativeStaking/CompoundingValidatorManager.sol

```diff
@@ -553,12 +575,16 @@ abstract contract CompoundingValidatorManager is Governable {
             withdrawalAddress
         );
 
+        // Store the validator state as verified
+        validator[pubKeyHash] = ValidatorData({
```

Comment:
why move this to before verifying the validator? It will be invalid if the withdrawal address does not match the strategy's address. Or is this to set the validatorIndex?